### PR TITLE
Profile page revamp

### PR DESCRIPTION
**Pull request by yksolanki9**
New design:
https://www.loom.com/share/a7fc9f9cc5934a52ab0269befb79fa65

Currency modal:
https://www.loom.com/share/7271dd004ddc405abf415a733dc08050

### CODE REVIEW

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<div class="employee-details-card">
+  <div class="employee-details-card__background">
+    <ion-grid class="employee-details-card__grid">
+      <ion-row>
+        <ion-col class="employee-details-card__icon-container">
+          <!-- <div class="employee-details-card__profile-icon">
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:6]: remove if not needed?

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    &__icon {
+      height: 100%;
+      background: $purple-gradient-1;
+      border-radius: 50%;
+    }
+
+    &__icon-border {
+      background: $purple-gradient-2;
+      border-radius: 50%;
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss:39]: this is common in &__icon as well, please reuse

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            </ion-row>
+            <ion-row class="employee-details-card__details-container">
+              <ion-col size="6.5" class="employee-details-card__mobile-container">
+                <div class="employee-details-card__label">Mobile</div>
+                <div class="display-flex employee-details-card__value">
+                  <div class="employee-details-card__mobile text-ellipsis">
+                    {{ eou.ou.mobile.length ? eou.ou.mobile : 'Add Number' }}
+                  </div>
+                  <ion-icon
+                    [src]="'../../../../assets/svg/' + (eou.ou.mobile ? 'fy-edit-gradient' : 'fy-plus') + '.svg'"
```
Comment (Dimple16) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:38]: this action icon should be `$brand-primary`

File: src/app/fyle/my-profile/my-profile.page.scss [hunk 4]
```diff
@@ -1,272 +1,65 @@
-$item-border-color: #e0e0e0;
-$menu-botton-color: #000;
-$background-color: #fff;
+@import '../../../theme/colors.scss';
+// $item-border-color: #e0e0e0;
```
Comment (Dimple16) [inline src/app/fyle/my-profile/my-profile.page.scss:2]: remove this and add it to the colors file

File: src/app/fyle/my-profile/preference-setting/preference-setting.component.scss [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    &__content {
+      font-size: 14px;
+      color: $black-light;
+      width: calc(100% - 76px);
+    }
+  }
+
+  &__content-container {
+    justify-content: space-between;
```
Comment (Dimple16) [inline src/app/fyle/my-profile/preference-setting/preference-setting.component.scss:22]: is this working as expected?

File: src/app/fyle/my-profile/preference-setting/preference-setting.component.scss [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    padding: 20px 16px;
+
+    &__title {
+      font-size: 18px;
+      font-weight: 500;
+      color: $pure-black;
+    }
+
+    &__content {
+      font-size: 14px;
```
Comment (Dimple16) [inline src/app/fyle/my-profile/preference-setting/preference-setting.component.scss:15]: this is by default

File: src/theme/colors.scss [hunk 7]
```diff
@@ -18,7 +18,6 @@ $highlight-black: #464d57 !default;
 $highlight-green: #66bb6a !default;
 $highlight-blue: #5c98e5 !default;
 $pink-shadow: #d7d7d799 !default;
-$brand-primary: #f36 !default;
```
Comment (Dimple16) [inline src/theme/colors.scss:21]: this was intentionally changed by @MasterYushi in his cta changes pr, please confirm this change

Comment (MasterYushi) [reply src/theme/colors.scss]: please use `$brand-primary` instead of `$primary`

File: src/app/core/services/tracking.service.ts [hunk 8]
```diff
@@ -880,4 +880,9 @@ export class TrackingService {
   viewReportInfo(properties) {
     this.eventTrack('View Report Info', properties);
   }
+
+  //Toggle settings
+  onToggleSetting(properties) {
```
Comment (abhishek1234321) [inline src/app/core/services/tracking.service.ts:885]: `onSettingToggle` or `onSettingsToggle`?

Comment (abhishek1234321) [reply src/app/core/services/tracking.service.ts:885]: not a P0, your call

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                  <div
+                    #mobileNumberRef="matTooltip"
+                    (click)="showTooltip(mobileNumberRef)"
+                    [matTooltip]="eou.ou.mobile"
+                    [matTooltipDisabled]="!isMobileNumberHidden"
+                    matTooltipPosition="below"
+                    matTooltipClass="tooltip"
+                    class="employee-details-card__mobile text-ellipsis"
+                  >
+                    {{ eou.ou.mobile && eou.ou.mobile.length ? eou.ou.mobile : 'Add Number' }}
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:35]: is this needed? `eou.ou.mobile` doesn't `eou` already have a model defined?

Comment (abhishek1234321) [reply src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:35]: the tooltip should come only when the length is exceeded

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss [hunk 10]
```diff
@@ -0,0 +1,76 @@
+@import '../../../../theme/colors.scss';
+
+.employee-details-card {
+  color: $pure-white;
+  font-size: 16px;
+  line-height: 1.3;
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss:6]: is this needed? global is 1.4 right?

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+  &__background {
+    background: url('../../../../assets/images/employee-card-bg.png') no-repeat bottom;
+    height: 164px;
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss:10]: is this needed?

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+  async updateMobileNumber() {
+    const updateMobileNumberPopover = await this.popoverController.create({
+      component: FyInputPopoverComponent,
+      componentProps: {
+        title: (this.eou.ou.mobile?.length ? 'Edit' : 'Add') + ' Mobile Number',
+        ctaText: 'Save',
+        inputLabel: 'Mobile Number',
+        inputValue: this.eou.ou.mobile,
+        inputType: 'tel',
```
Comment (abhishek1234321) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.ts:59]: check with @Dimple16 if this works fine in ios too?

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.ts (hunk 12 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/employee-details-card/employee-details-card.component.ts:59]: She has already tested. Working fine in ios

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html (hunk 9 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:35]: In some cases `eou.ou.mobile` is not assigned any value. So got an error `Cannot read property 'length' of undefined`. Added this as a failsafe

File: src/app/core/services/tracking.service.ts (hunk 8 above)
Comment (yksolanki9) [reply src/app/core/services/tracking.service.ts:885]: Going with `onSettingsToggle`

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss (hunk 10 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss:6]: Default is 20px. `line-height: 1.3` looks better

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss (hunk 11 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/employee-details-card/employee-details-card.component.scss:10]: If removed, it defaults to 158px. As per design it is 164px

**Pull request merged**